cibuild.cmd

```diff
@@ -156,6 +156,20 @@ if %ERRORLEVEL% NEQ 0 (
     goto :error
 )
 
+:: Only detect source control changes when running in the CI environment
+:: Detect if there are any changed files which should fail the build
+if DEFINED JENKINS_URL (
+    echo Detecting changed files...
+    git diff --cached HEAD --exit-code
```

Comment:
You could maybe do `git diff --exit-code HEAD` which should show all diffs between the work tree and HEAD.

Torn on `--quiet` versus `--exit-code`--it'd be nice to get a list on failure, so I guess I prefer `--exit-code`.